packages/opentelemetry-exporter-datadog/src/datadog.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as api from '@opentelemetry/api';
+import { ExportResult, NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { translateToDatadog } from './transform';
+import { AgentExporter, PrioritySampler, DatadogExporterConfig } from './types';
+import { URL } from 'url';
+
+/**
+ * Format and sends span information to Datadog Exporter.
+ */
+export class DatadogExporter implements SpanExporter {
+  private readonly _logger: api.Logger;
+  private readonly _exporter: typeof AgentExporter;
+  private _service_name: string;
+  private _env: string | undefined;
+  private _version: string | undefined;
+  private _tags: string | undefined;
+  private _url: string;
+  private _flushInterval: number;
+
+  constructor(config: DatadogExporterConfig = {}) {
+    this._url =
+      config.agent_url ||
+      process.env.DD_TRACE_AGENT_URL ||
+      'http://localhost:8126';
+    this._logger = config.logger || new NoopLogger();
+    this._service_name =
+      config.service_name || process.env.DD_SERVICE || 'dd-service';
+    this._env = config.env || process.env.DD_ENV;
+    this._version = config.version || process.env.DD_VERSION;
+    this._tags = config.tags || process.env.DD_TAGS;
+    this._flushInterval = config.flushInterval || 1000;
+    this._exporter = new AgentExporter(
+      { url: new URL(this._url), flushInterval: this._flushInterval },
+      new PrioritySampler()
+    );
+  }
+
+  /**
+   * Called to export sampled {@link ReadableSpan}s.
+   * @param spans the list of sampled Spans to be exported.
+   */
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ): void {
+    if (spans.length === 0) {
+      return resultCallback(ExportResult.SUCCESS);
+    }
+
+    const formattedDatadogSpans = translateToDatadog(
+      spans,
+      this._service_name,
+      this._env,
+      this._version,
+      this._tags
+    );
+
+    try {
+      this._exporter.export(formattedDatadogSpans);
```

Comment:
Is this sync?

Reply:
it is, but it's only appending a span to the `AgentExporter` buffer (which gets flushed on an interval), see: https://github.com/DataDog/dd-trace-js/blob/89e9caa74f643fa31e3aba80f608a0f8857f8b74/packages/dd-trace/src/exporters/agent/index.js#L18